postgre-client/src/tasks.rs

```diff
@@ -93,51 +97,80 @@ impl PgClient {
+                last_modified_at = tmp.last_modified_at, 
+                mutability = tmp.mutability, 
+                next_refresh_at = NOW + INTERVAL '1 day' 
+                FROM (
+        ",
+        );
 
-        query_builder.push_values(data, |mut b, key| {
-            let tsk = UrlWithStatus::new(key.metadata_url.as_str(), false); // status is ignoring here
-            b.push_bind(tsk.metadata_url);
-            b.push_bind(key.status);
-            b.push_bind(key.error);
+        query_builder.push_values(data, |mut b, task| {
+            let url = UrlWithStatus::new(task.metadata_url.as_str(), false); // status is ignored here
+            b.push_bind(url.metadata_url);
+            b.push_bind(task.status);
+            b.push_bind(task.etag.clone());
+            b.push_bind(task.last_modified_at);
+            b.push_bind(task.mutability);
         });
 
-        query_builder.push(") as tmp (tsk_metadata_url, tsk_status, tsk_error) WHERE tasks.tsk_metadata_url = tmp.tsk_metadata_url;");
+        query_builder.push(") as tmp (task_status, etag, last_modified_at, mutability) WHERE tasks.metadata_hash = tmp.metadata_hash;");
 
         let query = query_builder.build();
         query.execute(&self.pool).await?;
 
         Ok(())
     }
 
-    pub async fn get_pending_tasks(
+    pub async fn update_tasks_attempt_time(&self, data: Vec<String>) -> Result<(), IndexDbError> {
+        let mut query_builder: QueryBuilder<'_, Postgres> = QueryBuilder::new(
+            "
+            UPDATE tasks 
+            SET
+                next_refresh_at = NOW + INTERVAL '1 day' 
+                WHERE tasks.metadata_url IN (
+        ",
+        );
+
+        query_builder.push_values(data, |mut b, task| {
+            let url = UrlWithStatus::new(task.as_str(), false); // status is ignored here
+            b.push_bind(url.metadata_url);
+        });
+
+        query_builder.push(");");
+
+        let query = query_builder.build();
+        query.execute(&self.pool).await?;
+
+        Ok(())
+    }
+
+    pub async fn get_pending_metadata_tasks(
         &self,
         tasks_count: i32,
-    ) -> Result<Vec<JsonDownloadTask>, IndexDbError> {
-        let mut query_builder: QueryBuilder<'_, Postgres> =
-            QueryBuilder::new("WITH cte AS (
-                                        SELECT tsk_id
-                                        FROM tasks
-                                        WHERE (tsk_status = 'running' OR tsk_status = 'pending') AND tsk_locked_until < NOW() AND tsk_attempts < tsk_max_attempts
-                                        LIMIT ");
-
+    ) -> Result<Vec<MetadataDownloadTask>, IndexDbError> {
+        let mut query_builder: QueryBuilder<'_, Postgres> = QueryBuilder::new(
+            "WITH selected_tasks AS (
+                                    SELECT t.metadata_hash FROM tasks AS t
+                                    WHERE t.task_status = 'pending' AND NOW() > t.next_try_at
```

Comment:
May I suggest ordering those tasks? For example, we may want to process it in a fifo manner.